### PR TITLE
Allow setting the CA type when loading into cert manager and unloading specific CA types from the cert manager.

### PR DESCRIPTION
**Pull request by kareem-wolfssl**
# Description

Adds a new type WOLFSSL_USER_INTER, intended to be used to load intermediate certs.
This allows users to dynamically free certs marked with a certain type at runtime.

Fixes zd#20038

# Testing

Customer provided reproducer

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfssl/ssl.h [hunk 1]
```diff
@@ -4202,13 +4203,18 @@ WOLFSSL_API void wolfSSL_CTX_SetPerformTlsRecordProcessingCb(WOLFSSL_CTX* ctx,
 
     WOLFSSL_API int wolfSSL_CertManagerLoadCA(WOLFSSL_CERT_MANAGER* cm,
         const char* f, const char* d);
+    WOLFSSL_API int wolfSSL_CertManagerLoadCABuffer_ex2(WOLFSSL_CERT_MANAGER* cm,
```
Comment (JacobBarthelmeh) [inline wolfssl/ssl.h]: This would be the first _ex2 in wolfssl/ssl.h that I can see. Could we name it something like wolfSSL_CertMangerLoadCaBufferType ? That's personal preference though _ex2 also works.

File: wolfssl/ssl.h [hunk 2]
```diff
@@ -4202,13 +4203,18 @@ WOLFSSL_API void wolfSSL_CTX_SetPerformTlsRecordProcessingCb(WOLFSSL_CTX* ctx,
...
+        const unsigned char* buff, long sz, int format, int userChain,
+        word32 flags, int type);
     WOLFSSL_API int wolfSSL_CertManagerLoadCABuffer_ex(WOLFSSL_CERT_MANAGER* cm,
         const unsigned char* buff, long sz, int format, int userChain,
         word32 flags);
     WOLFSSL_API int wolfSSL_CertManagerLoadCABuffer(WOLFSSL_CERT_MANAGER* cm,
         const unsigned char* buff, long sz, int format);
 
     WOLFSSL_API int wolfSSL_CertManagerUnloadCAs(WOLFSSL_CERT_MANAGER* cm);
+    WOLFSSL_API int wolfSSL_CertManagerUnloadIntermediateCertsEx(
```
Comment (JacobBarthelmeh) [inline wolfssl/ssl.h]: Follow the `_ex` naming convention if adding this to ssl.h.

File: src/ssl_certman.c [hunk 3]
```diff
@@ -583,10 +591,59 @@ int wolfSSL_CertManagerLoadCABuffer_ex(WOLFSSL_CERT_MANAGER* cm,
         /* Clear certificate manager in WOLFSSL_CTX so it won't be freed. */
         tmp->cm = NULL;
     }
+    if (ret == WOLFSSL_SUCCESS && type != WOLFSSL_USER_CA) {
+        dCert = (DecodedCert*)XMALLOC(sizeof(DecodedCert), NULL,
```
Comment (JacobBarthelmeh) [inline src/ssl_certman.c]: Use the cm's heap hint for XMALLOC/XFREE of dCert.

**Review by JacobBarthelmeh — CHANGES_REQUESTED** [review]
A test case is need for each new API added. Please also test the boundaries of "type" added and how freeing signers handles other types.
